rustls/src/server/tls12.rs

```diff
@@ -393,16 +395,16 @@ mod client_hello {
         common.send_msg(c, false);
     }
 
-    fn emit_server_kx(
+    fn emit_server_kx<C: CryptoProvider>(
         transcript: &mut HandshakeHash,
         common: &mut CommonState,
         sigschemes: Vec<SignatureScheme>,
-        skxg: &'static kx::SupportedKxGroup,
+        skxg: &'static <<C as CryptoProvider>::KeyExchange as KeyExchange>::SupportedGroup,
```

Comment:
There are a number of places where the compiler forced me to disambiguate the traits but this is probably the grossest since we have to disambiguate the CryptoProvider to get the KeyExchange, and then the KeyExchange to get the SupportedGroup. 

It feels odd to me that there's ambiguity here but maybe this is common-place in this sort of Rust code?

Reply:
The type `C` could also implement another trait that also has a `KeyExchange` associated type, so Rust makes you spell it out to make sure to get the right one. It's verbose, but not much we can do about it.